ra/ra_test.go

```diff
@@ -1666,6 +1668,59 @@ func TestNewOrder_AuthzReuse_NoPending(t *testing.T) {
 	test.AssertNotEquals(t, new.V2Authorizations[0], extant.V2Authorizations[0])
 }
 
+func TestNewOrder_ProfileSelectionAllowList(t *testing.T) {
```

Comment:
This test is a good candidate for a parallel table-driven test with
```go
type testCase struct {
  list *allowlist.List
  wantErr string
}
```
You can also use `randomDomain()` for the dnsName in each iteration of the loop.